models/eventModel.js

```diff
@@ -67,12 +67,12 @@ const eventElement = new mongoose.Schema({
     oldLocation: {
         type: mongoose.Schema.Types.ObjectId,
         ref: 'Location',
-        required: {
-            validator: function () {
+        required: [
+            function () {
                 return this.change === "Move";
             },
-            message: "Location is required"
-        },
+            "Location is required"
+        ],
         validate: {
```

Comment:
Difference between required and validate ?

Reply:
Require built-in validator for checking presence if true,
Validate custom validator to verify properties of the data